libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -303,12 +303,16 @@ void ArrayUtils::copyArrayToMemory(ArrayType const& _sourceType, bool _padToWord
 			m_context << _sourceType.length();
 		if (baseSize > 1)
 			m_context << u256(baseSize) << Instruction::MUL;
-		// stack: target source_offset source_len
-		m_context << Instruction::DUP1 << Instruction::DUP3 << Instruction::DUP5;
-		// stack: target source_offset source_len source_len source_offset target
-		m_context << Instruction::CALLDATACOPY;
-		m_context << Instruction::DUP3 << Instruction::ADD;
-		m_context << Instruction::SWAP2 << Instruction::POP << Instruction::POP;
+
+		string routine = "calldatacopy(target, source, len)\n";
+		if (_padToWordBoundaries)
+			routine += R"(
+				mstore(add(target, len), 0)
```

Comment:
Why is this needed?

Reply:
To make sure that it actually ends in a sequence of zeros in case the memory was dirty.

Reply:
For a moment I thought this was mstore8 and that didn't make sense in terms of cleaning the entire slot. Still I'd think a short comment here wouldn't hurt explaining the intentions.